### ISSUE TITLE
Customizing host name

### ISSUE THREAD

**Issue by tobiasschittkowski** [IonxSolutions/serilog-sinks-syslog#33]
Hello,

is it possible to customize/change the hostname at setup?

Thanks!
Tobias

**Comment by colin-anderson** [comment]
Hi, if I understand correctly, you want to change the target hostname after the sink after it has already been configured? If my understanding is correct, that's not possible (in general, you cannot change Serilog sinks after they have been configured).

**Comment by tobiasschittkowski** [comment]
Hi, no - the data send to syslog contain a hostname, normally the name of the machine. I would like to change that name via configuration.

**Comment by colin-anderson** [comment]
Ah, understood now :smile:

At present, the only way to change this is to change the hostname of the actual machine the code is running on, as it will always use [Environment.MachineName](https://github.com/IonxSolutions/serilog-sinks-syslog/blob/master/src/Serilog.Sinks.Syslog/Sinks/Formatters/SyslogFormatterBase.cs#L25).

We could change the formatters so they could override the hostname, in the same way as we do for the application name, something like:

```c#
private readonly string host;

public Rfc3164Formatter(Facility facility = Facility.Local0, string applicationName = null, 
	MessageTemplateTextFormatter templateFormatter = null, string host = null)
	: base(facility, templateFormatter)
{
	this.applicationName = applicationName ?? ProcessName;
        this.host= host ?? Environment.MachineName;

	// Conform to the RFC
	this.applicationName = this.applicationName
		.AsPrintableAscii()
		.WithMaxLength(32);

	this.host = this.host.WithMaxLength(255);
}

// We'd need similar for Rfc5424Formatter too
```

Happy to take a PR for this if you fancy? (`SyslogRfc3164FormatterTests` and `SyslogRfc5424FormatterTests` would also need amended accordingy)

**Comment by tobiasschittkowski** [comment]
Sure, I'll give it a try!

**Comment by colin-anderson** [comment]
Merged and released in v2.0.5, thanks for contributing! ❤️
